### PR TITLE
Revamp Docker image build and push to support releases

### PR DESCRIPTION
**Pull request by Geod24**
With this change, the following will happen:
- The docker 'latest' tag, which is the default, will no longer be pushed to;
- When a PR is merged / something is pushed to a branch, 'devel' will be used instead;
- When a tag is pushed, the matching tag will be pushed to docker hub;
- When pushing a branch or a tag to one's fork, errors will no longer be generated;

The only downside here is to tie up the registry and agora build together,
so that if one fails, both have to be repeated. In practice, this almost never happened,
so it shouldn't be a great concern.

The goal is to latest trigger a push to 'latest' when a release is made.

### CODE REVIEW

File: .github/workflows/devel-docker.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+        # This step will checkout the source with all the submodules
+      - name: '[Agora] Build and push'
+        id: agore_build
+        uses: docker/build-push-action@v2
+        with:
+          push: ${{ steps.login.outcome == 'success' }}
+          tags: bpfk/agora:devel
```
Comment (hewison-chris) [inline .github/workflows/devel-docker.yml:39]: Just an idea: Couldn't we add a second tag which includes the github.sha so we can pull for testing.

Comment (Geod24) [reply .github/workflows/devel-docker.yml:39]: I think that would bloat the repository quite a bit. Plus, it's fairly easy to build locally and push, no ?

Comment (hewison-chris) [reply .github/workflows/devel-docker.yml:39]: Yes I suppose if you are testing your own build. If you want to share a pushed devel tag then I guess you can still add another tag manually.

**Review by hewison-chris — APPROVED** [review]
LGTM

**Pull request merged**